csrc/transformer/inference/csrc/apply_rotary_pos_emb.cu

```diff
@@ -194,6 +194,17 @@ __global__ void apply_rotary_pos_emb1(__half* mixed_query,
                                              : __shfl_sync(mask[lane], q_rot, lane - half_dim);
             auto k_rot_tmp = lane < half_dim ? __shfl_sync(mask[lane], k_rot, lane + half_dim)
                                              : __shfl_sync(mask[lane], k_rot, lane - half_dim);
+#if defined(__HIP_PLATFORM_HCC__)
+            auto q_rot_tmp = lane < half_dim ? __shfl(mask[lane], q_rot, lane + half_dim)
+                                             : __shfl(mask[lane], q_rot, lane - half_dim);
+            auto k_rot_tmp = lane < half_dim ? __shfl(mask[lane], k_rot, lane + half_dim)
+                                             : __shfl(mask[lane], k_rot, lane - half_dim);
```

Comment:
```suggestion
            auto q_rot_tmp = lane < half_dim ? __shfl(q_rot, lane + half_dim)
                                             : __shfl(q_rot, lane - half_dim);
            auto k_rot_tmp = lane < half_dim ? __shfl(k_rot, lane + half_dim)
                                             : __shfl(k_rot, lane - half_dim);
```
The function signature for __shfl is

`__device__ inline int __shfl(int var, int src_lane, int width = warpSize);`